### PR TITLE
Fixes issue 4922

### LINKED ISSUE #4922: Unable to restore app hooked pages on 3.2.x
PR #4883 introduced reversion 1.10.x support on the CMS.

Due to the way reversion 1.10.x works, the page object is restored and then saved using the change form logic but the PageAdmin's save_model assumes that the page has not yet been created and so tries to create a temporary page to get the treebeard attributes but this fails because things like application_namespace are unique at the database level.

- [x] Merge #4923 into 3.2.x

Fixed. To be forward ported to CMS 3.3.x

### PR DESCRIPTION
**Pull request by czpython**
Fixes #4922
* Adds test for regression.
* Fix is to create a "blank" page instead of a copy of the recovered page in order to avoid integrity errors.

### CODE REVIEW

File: cms/tests/test_reversion_tests.py [hunk 1]
```diff
@@ -231,6 +232,61 @@ def test_recover(self):
...
+        self.assertEqual(Page.objects.count(), 2)
+
+        with self.login_user_context(self.user):
+
+            page_data = self.get_new_page_data_dbfields()
+            page_data['apphook'] = 'SampleApp'
+            page_data['apphook_namespace'] = 'SampleApp'
+
+            with create_revision():
+                # Need to create page manually to add apphooks
```
Comment (yakky) [inline cms/tests/test_reversion_tests.py]: why not using the API?

**Comment by yakky** [comment]
Apart for the clarification LGTM

File: cms/tests/test_reversion_tests.py (hunk 1 above)
Comment (czpython) [inline cms/tests/test_reversion_tests.py]: I am using API below in ``page = create_page(**page_data)``.
that's what I meant by "manually".

Comment (yakky) [inline cms/tests/test_reversion_tests.py]: Why do I do review in the evening?
Disregard my comment

**Pull request merged**